src/_pytest/fixtures.py

```diff
@@ -240,6 +241,24 @@ def getfixturemarker(obj: object) -> Optional["FixtureFunctionMarker"]:
 _Key = Tuple[object, ...]
 
 
+@attr.s(auto_attribs=True, eq=False)
+class SafeHashWrapper:
+    obj: Any
+
+    def __eq__(self, other) -> Any:
+        try:
+            res = self.obj == other
+            bool(res)
+            return res
+        except Exception:
+            return id(self.obj) == id(other)
+
+    def __hash__(self) -> Any:
+        if isinstance(self.obj, Hashable):
+            return hash(self.obj)
+        return hash(id(self.obj))
```

Comment:
Some suggestions here (I haven't checked that they all work):

```suggestion
@attr.s(auto_attribs=True, eq=False, slots=True)
class SafeHashWrapper:
    obj: object

    def __eq__(self, other: object) -> bool:
        try:
            res = self.obj == other
            return bool(res)
        except Exception:
            return id(self.obj) == id(other)

    def __hash__(self) -> int:
        if isinstance(self.obj, Hashable):
            return hash(self.obj)
        return hash(id(self.obj))
```

Reply:
Thank's, applied all except `obj: Any`, because it makes mypy think `obj` is always hashable and therefore we have an unreachable statement in `SafeHashWrapper.__hash__`.

I think the type constraint I'd like to express for `obj` is "type supports identity" and "type is not NoneType". But I don't know a way how, do you?

A bit more OT, if `isinstance(object(), Hashable)` and `isinstance(dict(), object)` are `True`, doesn't `isinstance(dict(), Hashable) == False` break Liskov?